backend/ipmi.pm

```diff
@@ -32,7 +32,11 @@ sub ipmitool ($self, $cmd, %args) {
     my @tries = (1 .. $args{tries});
     for (@tries) {
         $ret = IPC::Run::run(\@cmd, \$stdin, \$stdout, \$stderr);
-        if ($ret == 0) {
+        bmwqemu::diag("IPMI: IPC::Run::run return $ret");
+        # Refer to https://metacpan.org/pod/IPC::Run#RETURN-VALUES
+        # "run() and finish() return TRUE when all subcommands exit with a 0 result code."
+        # "This is the opposite of perl's system() command."
+        if ($ret) {
```

Comment:
Not sure whether we really need to quote the documentation here. If we would do that on any use of that function or even other functions it would clutter our code a lot. So maybe let's avoid setting a bad precedence here. You could still add a short inline comment:

```suggestion
        if ($ret) {  # command succeeded
```

I would also avoid printing the return value literally as it is just a boolean and not really intended for literal printing. To be able to follow whether the dell-sleep path was entered I'd move the logging into the function itself.

Reply:
Thanks and agree! The code is updated already.